rotkehlchen/exchanges/bitstamp.py

```diff
@@ -282,10 +295,17 @@ def query_online_deposits_withdrawals(
         # they correspond to so we can also have the fee taken into account
         indices_to_delete = []
         for asset_movement in asset_movements:
+            assert asset_movement.extra_data is not None  # should always be set here
```

Comment:
We try to avoid assertions in the code. Prefer type ignore.

But this shows something weird. Isn't `extra_data` always set in one way or another for asset movements? If so the class should have it as non optional so all these type checks should succeed and not need type ignore or assets. It will be much better if you can make it so that it's set always if this is what happens.

Same for all other places (like right below) where you do this in this PR and probably in other parts of the code.

Reply:
> Isn't extra_data always set in one way or another for asset movements?

No, `extra_data` is not always set. For instance fees generally have no `extra_data` set, and even normal deposit/withdrawals are not guaranteed to have anything in `extra_data`. Which actually means that it needs to be doing an if check on `asset_movement.extra_data` here to skip the fee events, so I've added that.  `crypto_movement.extra_data` should always be set though, so I changed that to use type ignore.